analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs

```diff
@@ -126,8 +126,8 @@ private void ReportUnusedParametersOnMethod(MethodContext declaration)
 
         private void ReportOnDeadParametersAtEntry(MethodContext declaration, IImmutableList<IParameterSymbol> noReportOnParameters)
         {
-            var bodyNode = (CSharpSyntaxNode)declaration.Body ?? declaration.ExpressionBody;
-            if (bodyNode == null || declaration.Context.Node.IsKind(SyntaxKind.ConstructorDeclaration))
+            var declarationNode = (CSharpSyntaxNode)declaration.MethodDeclaration ?? declaration.LocalFunctionDeclaration;
```

Comment:
For simplicity, there's just one node around here now
```suggestion
            var node = (CSharpSyntaxNode)declaration.MethodDeclaration ?? declaration.LocalFunctionDeclaration;
```